src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+package mate.academy.dao.impl;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connectionproperties.ConnectionProperties;
+import mate.academy.dao.BookDao;
+import mate.academy.entity.Book;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    public static final String INSERT_BOOK_QUERY = "INSERT INTO books (title,price) values (?,?)";
+    public static final String SELECT_FROM_BOOKS_WHERE_ID = "SELECT * FROM books WHERE id = ?";
+    public static final String SELECT_FROM_BOOKS = "SELECT * FROM books";
+    public static final String UPDATE_BOOKS_QUERY = "UPDATE books SET title=?, price=? where id=?";
+    public static final String DELETE_FROM_BOOKS_WHERE_ID = "DELETE FROM books WHERE id = ?";
+    public static final String EXPECTED_CREATE = "Expected crate at least one book, but created ";
+    public static final String CAN_T_SAVE_A_BOOK_MSG = "Can't save a book ";
+    public static final String CAN_T_GET_A_BOOK_BY_ID_MSG = "Can't get a book by id ";
+    public static final String CAN_T_GET_A_LIST_OF_BOOK_MSG = "Can't get a list of book ";
+    public static final int PARAMETER_INDEX_ONE = 1;
+    public static final int PARAMETER_INDEX_TWO = 2;
+    public static final int COLUMN_INDEX_THREE = 3;
+    public static final int MIN_UPDATE_BOOK = 1;
+    public static final List<Book> BOOK_LIST = new ArrayList<>();
```

Comment:
The `BOOK_LIST` is declared as a static field, which means it will retain its state across multiple invocations of `findAll()`. This could lead to incorrect results if the list is not cleared before adding new elements. Consider using a local list inside the `findAll()` method instead.